lang/java/reef-runtime-yarn/src/main/java/org/apache/reef/runtime/yarn/driver/restart/DFSEvaluatorLogAppendReaderWriter.java

```diff
@@ -65,6 +66,11 @@ public synchronized void writeToEvaluatorLog(final String formattedEntry) throws
     }
   }
 
+  @Override
+  public Iterable<String> readFromEvaluatorLog() throws IOException {
```

Comment:
Is your concern that the user might not complete the iteration? Internally, when the iteration completes and there are no more elements to read, the reader closes the DFS file.